File: CurrencyConverter/app/src/main/java/com/nsh/currencyconverter/views/adapters/OnboardingViewPagerAdapter.java
package com.nsh.currencyconverter.views.adapters;
import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;
import com.nsh.currencyconverter.views.fragments.OnboardingFragment1;
import com.nsh.currencyconverter.views.fragments.OnboardingFragment2;
import com.nsh.currencyconverter.views.fragments.OnboardingFragment3;

public class OnboardingViewPagerAdapter extends FragmentStateAdapter {

    public OnboardingViewPagerAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        switch (position) {
            case 1:
                return new OnboardingFragment2();
            case 2:
                return new OnboardingFragment3();
            default:
                return new OnboardingFragment1();
        }
    }

    @Override
    public int getItemCount() {
        return 3;
    }
}

File: CurrencyConverter/app/src/main/java/com/nsh/currencyconverter/views/adapters/OnboardingViewPagerAdapter.kt
package com.nsh.currencyconverter.views.adapters

import androidx.annotation.NonNull
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.nsh.currencyconverter.views.fragments.OnboardingFragment1
import com.nsh.currencyconverter.views.fragments.OnboardingFragment2

class OnboardingViewPagerAdapter(fragmentActivity: FragmentActivity) : FragmentStateAdapter(fragmentActivity) {

    @NonNull
    override fun createFragment(position: Int): Fragment {
        return when (position) {
            1 -> OnboardingFragment2()
            else -> OnboardingFragment1()
        }
    }

    override fun getItemCount(): Int {
        return 3
    }
}
